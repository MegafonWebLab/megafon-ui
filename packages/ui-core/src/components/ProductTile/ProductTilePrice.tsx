import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './style/ProductTilePrice.less';
import AnimationValue from './ProductTileValue';

interface IProductTilePriceProps {
    /** title */
    title?: string;
    /** value */
    value: string;
    /** unitExtra */
    unitExtra: string;
    /** unitValue */
    unitValue: string;
    /** discount */
    discount?: string;
}

const cn = cnCreate('mfui-product-tile-price');
class ProductTilePrice extends React.Component<IProductTilePriceProps> {
    static propTypes = {
        title: PropTypes.string,
        value: PropTypes.string.isRequired,
        unitExtra: PropTypes.string,
        unitValue: PropTypes.string,
        discount: PropTypes.string,
    };

    render() {
        const { title, value, unitExtra, unitValue, discount } = this.props;
        const valueElem = <AnimationValue value={(discount || value) + ` ${unitValue}`} />;
        const unitElem = <span className={cn('payment-period')}>{unitExtra}</span>;

        return (
            <div className={cn('', { discount: !!discount })}>
                <span className={cn('old-price')}>
                    <span className={cn('old-price-value')}>
                        {discount ? `${value} ${unitValue}` : ' '}
                    </span>
                </span>
                <span className={cn('actual-price')}>
                    {valueElem} {unitElem}
                </span>
                <div className={cn('discount-condition')}>{title}</div>
            </div>
        );
    }
}

export default ProductTilePrice;
