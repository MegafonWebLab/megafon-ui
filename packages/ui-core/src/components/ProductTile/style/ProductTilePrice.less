@import '~styles/base.less';

@b: e('.mfui-product-tile-price');

@{b} {
    .h2();
    text-align: center;

    @media @desktopS {
        margin-bottom: 1px;
    }

    @media @desktopMU {
        margin-bottom: 3px;
    }

    &__payment-period {
        font-weight: normal;
        font-size: 15px;
        line-height: 24px;
        letter-spacing: 0;
    }

    &__old-price {
        visibility: hidden;
        position: relative;
        display: inline-block;
        font-size: 15px;
        margin-right: 6px;
        line-height: 24px;
        font-weight: 500;
        letter-spacing: 0.5px;

        @media @mobileB {
            font-size: 18px;
            margin: 0 auto;
        }

        @media @desktopS {
            font-size: 18px;
        }

        @media @desktopMU {
            font-size: 20px;
            line-height: 28px;
            margin-right: 11px;
        }

        &:before {
            content: '';
            position: absolute;
            width: 100%;
            height: 2px;
            margin: auto;
            top: 0;
            bottom: 0;
            background-color: @fury;
        }
    }

    &_discount &__old-price {
        visibility: visible;
    }

    &__old-price-value {
        opacity: 0.5;
    }

    &__actual-price {
        @media @mobileB {
            display: block;
            text-align: center;
        }
    }

    &__discount-condition {
        font-size: 12px;
        font-weight: normal;
        line-height: 16px;
        min-height: 16px;
    }
}
