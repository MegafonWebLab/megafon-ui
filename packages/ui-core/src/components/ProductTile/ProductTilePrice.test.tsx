import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTilePrice from './ProductTilePrice';

const props = {
    title: 'title',
    value: '23423',
    unitExtra: 'uniteExtra',
    unitValue: 'uniteValue',
};

describe('<ProductTilePrice />', () => {
    it('renders ProductTilePrice', () => {
        const wrapper = shallow(<ProductTilePrice {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTilePrice with discount', () => {
        const wrapper = shallow(<ProductTilePrice {...props} discount="100" />);
        expect(wrapper).toMatchSnapshot();
    });
});
