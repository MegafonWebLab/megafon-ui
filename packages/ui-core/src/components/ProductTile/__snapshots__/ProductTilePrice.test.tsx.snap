// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTilePrice /> renders ProductTilePrice 1`] = `
<div
  className="mfui-product-tile-price"
>
  <span
    className="mfui-product-tile-price__old-price"
  >
    <span
      className="mfui-product-tile-price__old-price-value"
    >
       
    </span>
  </span>
  <span
    className="mfui-product-tile-price__actual-price"
  >
    <ProductTileValue
      value="23423 uniteValue"
    />
     
    <span
      className="mfui-product-tile-price__payment-period"
    >
      uniteExtra
    </span>
  </span>
  <div
    className="mfui-product-tile-price__discount-condition"
  >
    title
  </div>
</div>
`;

exports[`<ProductTilePrice /> renders ProductTilePrice with discount 1`] = `
<div
  className="mfui-product-tile-price mfui-product-tile-price_discount"
>
  <span
    className="mfui-product-tile-price__old-price"
  >
    <span
      className="mfui-product-tile-price__old-price-value"
    >
      23423 uniteValue
    </span>
  </span>
  <span
    className="mfui-product-tile-price__actual-price"
  >
    <ProductTileValue
      value="100 uniteValue"
    />
     
    <span
      className="mfui-product-tile-price__payment-period"
    >
      uniteExtra
    </span>
  </span>
  <div
    className="mfui-product-tile-price__discount-condition"
  >
    title
  </div>
</div>
`;
